pkg/consts/dataplane.go

```diff
@@ -178,3 +178,13 @@ const (
 	// to ensure that the resources are not deleted before the DataPlane is deleted.
 	DataPlaneOwnedWaitForOwnerFinalizer = "gateway-operator.konghq.com/wait-for-owner"
 )
+
+// -----------------------------------------------------------------------------
+// Consts - DataPlaneKonnectExtension Finalizers
+// -----------------------------------------------------------------------------
+
+const (
+	// DataPlaneExtensionFinalizer is the finalizer added to DataPlaneKonnectExtension
+	// to ensure that the resources are not deleted when in use by a dataplane.
+	DataPlaneExtensionFinalizer = "gateway-operator.konghq.com/extension-in-use"
```

Comment:
Is the use case for this finalizer simply to block deletion of the extension when referenced by a DP? I'm wondering what are we gaining with this vs allowing them to be deleted and reconciling the DP accordingly (with a proper status condition indicating such a situation). 🤔 

LMKWYT

Reply:
This extension completely changes the behavior and the configuration of a DataPlane. From the user perspective, I'd be baffled to see my DataPlane entirely reconfigured with a completely different setup when deleting a resource. I strongly think that we should block the deletion of an extension in use, to avoid the above.

Reply:
Thanks for explanation. Now this makes more sense to me. 👍